### DISCUSSION TITLE
Xamarin.Forms iOS Reducing App Size

### DISCUSSION THREAD

**Discussion by AlonRom** [xamarin/Xamarin.Forms#11318] [General]
I received a size limit warning when releasing Xamarin.Forms app for iOS.

Why is the option to "Enable Bitcode" is not available in Xamarin.Forms?

what is the suggested way to configure the Build when releasing a Xamarin.Forms app targeting iOS?

What are the advantages and disadvantages of using:

"Use LLVM optimizing compiler"
Use Thumb-2 instruction set for ARMv7 and ARMv7s

**Comment by davidortinau** [comment]
Bitcode is used for watchOS and tvOS. We have not implemented it for Xamarin.iOS, and hence it's not available for Xamarin.Forms iOS apps. 

The suggested build profile is what you get by default when selecting "Release" mode, however there are many variables as the screen shows and so if the default isn't meeting your needs you'll want to study the options and proceed to experiment.

LLVM - https://developer.apple.com/library/archive/documentation/CompilerTools/Conceptual/LLVMCompilerOverview/
- can produce smaller and faster code, but not always. My app actually gained a few KB last I tried.

I recommend spending some time in this section of the docs to learn more about each option: https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/

I also just last week archived for publishing from Visual Studio and then used Xcode to sign and publish the archive using their app-thinning process (since we generate native code, it's just like an app created in Xcode at that point). My iOS Xamarin.Forms app was a hair over 9mb, very light on assets so basically a "hello world".

**Reply by AlonRom** [reply]
@davidortinau thanks for your response.

- Is "Enable Bitcode" planned for the near future? currently, the size (on the same app) on iOS is significantly larger than Android.

- Will enabling "Use LLVM optimizing compiler" reduce my app size, do you advise to enable it?

- Will enabling "Use Thumb-2 instruction set for ARMv7 and ARMv7s" reduce my app size, do you advise to enable it?


Thanks.
